pack.go

```diff
@@ -90,7 +118,61 @@ func Pack(ctx context.Context, pusher content.Pusher, layers []ocispec.Descripto
 		Size:      int64(len(manifestBytes)),
 	}
 
-	// push manifest.
+	// push manifest
+	if err := pusher.Push(ctx, manifestDesc, bytes.NewReader(manifestBytes)); err != nil && !errors.Is(err, errdef.ErrAlreadyExists) {
+		return ocispec.Descriptor{}, fmt.Errorf("failed to push manifest: %w", err)
+	}
+
+	return manifestDesc, nil
+}
+
+// PackArtifact packs the given blobs, generates an ORAS Artifact Manifest for
+// the pack, and pushes it to a content storage.
+// If succeeded, returns a descriptor of the manifest.
+// Returns ErrMissingArtifactType if artifactType is empty.
+// Reference: https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md
+func PackArtifact(ctx context.Context, pusher content.Pusher, artifactType string, blobs []artifactspec.Descriptor, opts PackArtifactOptions) (ocispec.Descriptor, error) {
+	if artifactType == "" {
+		// artifactType is required for ORAS Artifact Manifest
+		return ocispec.Descriptor{}, ErrMissingArtifactType
+	}
+
+	// copy the original annotation map
+	annotations := make(map[string]string, len(opts.ManifestAnnotations))
+	for k, v := range opts.ManifestAnnotations {
+		annotations[k] = v
+	}
+	_, ok := annotations[AnnotationArtifactCreated]
+	if !ok {
+		// set creation time in RFC3339 format, if not provided
+		// reference: https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties
+		now := time.Now().UTC()
+		annotations[AnnotationArtifactCreated] = now.Format(time.RFC3339)
+	}
+
+	if blobs == nil {
+		blobs = []artifactspec.Descriptor{} // make it an empty array to prevent potential server-side bugs
+	}
```

Comment:
Wondering if this is really needed. 
As per [the artifact spec](https://github.com/oras-project/artifacts-spec/blob/main/artifact-manifest.md#oras-artifact-manifest-properties), `blobs` is an OPTIONAL collection of 0 or more blobs.
Since it is optional, `null` should be acceptable. Just that an empty array might be safer.